pubsub/common.go

```diff
@@ -0,0 +1,29 @@
+package pubsub
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/go-redis/redis/v8"
+)
+
+// CreateStream tries to create stream with given name, if it already exists
+// does not return an error.
+func CreateStream(ctx context.Context, streamName string, client redis.UniversalClient) error {
+	_, err := client.XGroupCreateMkStream(ctx, streamName, streamName, "$").Result()
+	if err == nil || err.Error() == "BUSYGROUP Consumer Group name already exists" {
```

Comment:
This can be more generic by checking if stream exists instead of looking for one specific error:
```
if err == nil { return nil}
if ctx.Error() != nil {return ctx.Error}
if StreamExists(...) {return nil}
return err
```

Reply:
Done.